From 55b6e729f33e38a372861fa630160e8af4c0ea33 Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <tlanger@maxlinear.com>
Date: Thu, 8 Jul 2021 18:52:43 +0200
Subject: [PATCH] PONRTSYS-9223: xrx500_phy_fw: Update kernel checks for PRX B2
 device

Correct some checks that were wrong for newer chip revisions.
---
 drivers/net/ethernet/lantiq/xrx500_phy_fw.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
index 618f7ee437ff..5005636e6dd3 100644
--- a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
+++ b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
@@ -238,7 +238,7 @@ static u32 prx300_gphy_config_rcal_rcm(struct xway_gphy_data *priv)
 	u32 val;
 	int retry;
 
-	if (ltq_get_soc_rev() == 1) {
+	if (ltq_get_soc_rev()) {
 		/* B-Step: get rcal from cdb register */
 		val = gsw_reg_r32(priv->fcsi_base,
 				  PRX300_GPHY_CDB_FCSI_PLL_RCALSTAT) & 0xF;
@@ -549,10 +549,10 @@ static int prx300_dt_parse(struct xway_gphy_data *priv)
 	}
 
 	/* Check SoC version and set firmware name accordingly */
-	if (ltq_get_soc_rev() == 1)
-		priv->fw_name = FIRMWARE_P31G_PRX300_B;
-	else
+	if (ltq_get_soc_rev() == 0)
 		priv->fw_name = FIRMWARE_P31G_PRX300_A;
+	else
+		priv->fw_name = FIRMWARE_P31G_PRX300_B;
 
 	return 0;
 }
